accord-core/src/main/java/accord/api/ConfigurationService.java

```diff
@@ -0,0 +1,56 @@
+package accord.api;
+
+import accord.local.Node;
+import accord.topology.Topology;
+
+/**
+ * Cluster configuration service. Manages linearizable cluster configuration changes. Any node reporting a
+ * configuration with epoch n is guaranteed to have previously seen applied every previous epoch configuration
+ */
+public interface ConfigurationService
+{
+    interface Listener
+    {
+        /**
+         * Informs listeners of new topology. This is guaranteed to be called sequentially for each epoch after
+         * the initial topology returned by `currentTopology` on startup.
+         */
+        void onTopologyUpdate(Topology topology);
+
+        /**
+         * Called when accord data associated with a superseded epoch has been sync'd across current replicas
+         */
+        void onEpochSyncComplete(Node.Id node, long epoch);
```

Comment:
Maybe add some additional explanation about what implementations need to guarantee before they invoke this?